sklearn/model_selection/_plot.py

```diff
@@ -118,9 +124,26 @@ def plot(
             The type of score to plot. Can be one of `"test"`, `"train"`, or
             `"both"`.
 
-        log_scale : bool, default=False
+        log_scale : bool, default="deprecated"
             Whether or not to use a logarithmic scale for the x-axis.
 
+            .. deprecated:: 1.3
+               `log_scale` is deprecated in 1.3 and will be removed in 1.5.
+               Use `xscale` and `yscale` instead.
+
+        xscale : {"linear", "log"}, default=None
+            The x-axis scale of the plot. `None` is equivalent to `"linear"` during
+            the deprecation period of `log_scale`.
+
+            .. versionadded:: 1.3
+            .. versionchanged:: 1.5
+               The default of `xscale` will change from `None` to `"linear"`.
+
+        yscale : {"linear", "log"}, default="linear"
+            The y-axis scale of the plot.
+
+            .. versionadded:: 1.3
```

Comment:
If we're not introducing the "auto" option that we discussed irl for xscale, I think we should just deprecate log_scale and let the user customize `ax_` himself.

Thinking more about that "auto" option, I'm not sure it's such a good idea. The threshold to switch from linear to log is arbitrary and might be deceiving for a bunch of users. We usually don't do that kind of auto-magic configuration.